include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -133,52 +248,76 @@ namespace shapes
                 t = clamp( t, 0.0, 1.0 );
 
                 // 1 root
-                float2 qos = c + (2.0*a + b*t)*t;
-                res = float2( length(qos),t);
+                float2_t qos = _C + (B + A*t)*t;
+                res = float2_t( length(qos),t);
             }
             else
             {
-                float z = sqrt(-p);
-                float v = acos( q/(p*z*2.0) ) / 3.0;
-                float m = cos(v);
-                float n = sin(v)*1.732050808;
-                float3 t = float3(m + m, -n - m, n - m) * z - kx;
+                float_t z = sqrt(-p);
+                float_t v = acos( q/(p*z*2.0) ) / 3.0;
+                float_t m = cos(v);
+                float_t n = sin(v)*1.732050808;
+                float3_t t = float3_t(m + m, -n - m, n - m) * z - kx;
                 t = clamp( t, 0.0, 1.0 );
 
                 // 3 roots
-                float2 qos = c + (2.0*a + b*t.x)*t.x;
-                float dis = dot(qos,qos);
+                float2_t qos = _C + (B + A*t.x)*t.x;
+                float_t dis = dot(qos,qos);
                 
-                res = float2(dis,t.x);
+                res = float2_t(dis,t.x);
 
-                qos = c + (2.0*a + b*t.y)*t.y;
+                qos = _C + (B + A*t.y)*t.y;
                 dis = dot(qos,qos);
-                if( dis<res.x ) res = float2(dis,t.y );
+                if( dis<res.x ) res = float2_t(dis,t.y );
 
-                qos = c + (2.0*a + b*t.z)*t.z;
+                qos = _C + (B + A*t.z)*t.z;
                 dis = dot(qos,qos);
-                if( dis<res.x ) res = float2(dis,t.z );
+                if( dis<res.x ) res = float2_t(dis,t.z );
 
                 res.x = sqrt( res.x );
             }
             
-            return res.x;
+            return res;
         }
-
-        float signedDistance(float2 pos)
+        
+        float_t signedDistance(float2_t pos, float_t thickness)
         {
-            return abs(ud(pos)) - thickness;
+            return abs(ud(pos)).x - thickness;
         }
-    };
+        
+        float_t calcArcLenInverse(float_t arcLen, float_t accuracyThreshold, float_t hint, ArcLengthPrecomputedValues preCompValues)
+        {
+            float_t xn = hint;
+
+            if (arcLen <= accuracyThreshold)
+                return arcLen;
+
+                // TODO: implement halley method
+            const int iterationThreshold = 32;
+            for(int n = 0; n < iterationThreshold; n++)
+            {
+                float_t arcLenDiffAtParamGuess = arcLen - calcArcLen(xn, preCompValues);
 
+                if (abs(arcLenDiffAtParamGuess) < accuracyThreshold)
+                    return xn;
+
+                float_t differentialAtGuess = length(2.0*A * xn + B);
+                    // x_n+1 = x_n - f(x_n)/f'(x_n)
+                xn -= (calcArcLen(xn, preCompValues) - arcLen) / differentialAtGuess;
+            }
+
+            return xn;
+        }
+    };
+    
     struct CubicBezier
```

Comment:
remove the `CubicBezier` completely we don't work with cubics anymore

Reply:
done